pkg/utils/kubelessutil.go

```diff
@@ -0,0 +1,36 @@
+package utils
+
+import (
+	"os"
+
+	clientsetAPIExtensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+)
+
+// GetConfigLocation returns a map which has information on the namespace where Kubeless controller is installed and the name of the ConfigMap which stores kubeless configurations
+func GetConfigLocation(apiExtensionsClientset clientsetAPIExtensions.Interface) (map[string]string, error) {
+	configLocation := make(map[string]string)
+	controllerNamespace := os.Getenv("KUBELESS_NAMESPACE")
+	kubelessConfig := os.Getenv("KUBELESS_CONFIG")
+
+	annotationsCRD, err := GetAnnotationsFromCRD(apiExtensionsClientset, "functions.kubeless.io")
+	if err != nil {
+		return nil, err
+	}
+	if len(controllerNamespace) == 0 {
+		if ns, ok := annotationsCRD["kubeless.io/namespace"]; ok {
+			controllerNamespace = ns
+		} else {
+			controllerNamespace = "kubeless"
+		}
+	}
+	configLocation["namespace"] = controllerNamespace
+	if len(kubelessConfig) == 0 {
+		if config, ok := annotationsCRD["kubeless.io/config"]; ok {
+			kubelessConfig = config
+		} else {
+			kubelessConfig = "kubeless-config"
+		}
+	}
+	configLocation["name"] = kubelessConfig
+	return configLocation, nil
```

Comment:
Just to resume our previous conversation:

> Minor comment: we can simply return two strings (and document the order), if not users of the function should know somehow how to use the proper keys

> sayanh 22 hours ago  Contributor
Sure but this is implemented exactly to avoid additional documentation as the code makes it self explanatory.

Yes, the code is self-explanatory, what I meant with the documentation was to say the order of the returned parameters in order to say that the first element was the namespace and then the name or vice versa.

Actually, a better solution would be to return a struct (instead of a map). That way it is not necessary to document anything and it is clear that `configLocation` has only two keys and their meaning:

```
type configLocation struct {
  Name string
  Namespace string
}
```

Reply:
Cool, I changed `ConfigLocation` to a struct